doc: Add default values to all properties

akien-mga opened the pull request:
Thanks to @bojidar-bg's impressive work in #29380.

I'll add another commit to remove all the manually specified default values.

akien-mga committed:
doc: Add default values to all properties

Thanks to @bojidar-bg's impressive work in #29380.

akien-mga commented on doc/classes/AnimationPlayer.xml:
```diff
@@ -223,37 +223,37 @@
 			The call mode to use for Call Method tracks. Default value: [constant ANIMATION_METHOD_CALL_DEFERRED].
 		</member>
-		<member name="playback_active" type="bool" setter="set_active" getter="is_active">
+		<member name="playback_active" type="bool" setter="set_active" getter="is_active" default="null">
```
This one seems bogus @bojidar-bg.

bojidar-bg replied:
Working on it!

bojidar-bg replied:
Done: #30187.

akien-mga commented on doc/classes/CPUParticles.xml:
```diff
@@ -28,188 +28,188 @@
 			If [code]true[/code], only one emission cycle occurs. If set [code]true[/code] during a cycle, emission will stop at the cycle's end. Default value: [code]false[/code].
 		</member>
-		<member name="orbit_velocity" type="float" setter="set_param" getter="get_param">
+		<member name="orbit_velocity" type="float" setter="set_param" getter="get_param" default="null">
```
This one and `orbit_velocity_random` below should not be `null` but `0`. This probably happens because those parameters are only used when `disable_z` is true, which is not the default for (3D) CPUParticles.

akien-mga commented on doc/classes/CPUParticles2D.xml:
```diff
@@ -29,183 +29,183 @@
 			If [code]true[/code], results in fractional delta calculation which has a smoother particles display effect. Default value: [code]true[/code].
 		</member>
-		<member name="gravity" type="Vector2" setter="set_gravity" getter="get_gravity">
+		<member name="gravity" type="Vector2" setter="set_gravity" getter="get_gravity" default="Vector2( 0, 98.8 )">
```
Note to self: review this, doesn't seem to match 10 times the standard gravity: https://en.wikipedia.org/wiki/Standard_gravity

Xrayez replied:
I always set default gravity to `Vector2(0, 700)`, higher than that the objects tend to penetrate colliders more often, but that could be solved with higher linear damp perhaps.

cbscribe replied:
Since 2D units are pixels, there's no relation to m/s^2 anyway. 98 is just a really low value.

akien-mga replied:
Well, I didn't say it should change, just that 98.8 != 98.07 which would be 10g. The 2D physics engine considers 10px = 1m by default, hence this default value.

akien-mga commented on doc/classes/CanvasItem.xml:
```diff
@@ -511,28 +511,28 @@
 			If [code]true[/code], the object draws behind its parent. Default value: [code]false[/code].
 		</member>
-		<member name="show_on_top" type="bool" setter="_set_on_top" getter="_is_on_top">
+		<member name="show_on_top" type="bool" setter="_set_on_top" getter="_is_on_top" default="null">
```
Also bogus, this should be a boolean.

akien-mga replied:
We can probably grep for `type="bool".*default="null"` to find other similar occurrences (and same with other non-Object types).

akien-mga commented on doc/classes/ConeTwistJoint.xml:
```diff
@@ -13,23 +13,23 @@
 			The ease with which the joint starts to twist. If it's too low, it takes more force to start twisting the joint.
 		</member>
-		<member name="swing_span" type="float" setter="_set_swing_span" getter="_get_swing_span">
+		<member name="swing_span" type="float" setter="_set_swing_span" getter="_get_swing_span" default="45.0">
```
For this and `twist_span`, the default value was given in radians. Should be checked.

bojidar-bg replied:
The getter and setter automatically convert to/from radians though https://github.com/godotengine/godot/blob/cd423f47128003d03b94b85a89753bb97e38870c/scene/3d/physics_joint.cpp#L513-L521

akien-mga replied:
Ok, so the default value should be correct and the docs were wrong.

akien-mga commented on doc/classes/Line2D.xml:
```diff
@@ -66,37 +66,37 @@
 			The texture used for the line's texture. Uses [code]texture_mode[/code] for drawing style.
 		</member>
-		<member name="texture_mode" type="int" setter="set_texture_mode" getter="get_texture_mode" enum="Line2D.LineTextureMode">
+		<member name="texture_mode" type="int" setter="set_texture_mode" getter="get_texture_mode" enum="Line2D.LineTextureMode" default="1101004800">
```
Looks bogus, uninitalized variable?

bojidar-bg replied:
Indeed, can you fix it? ([line_2d.cpp:41](https://github.com/godotengine/godot/blob/master/scene/2d/line_2d.cpp#L41), `_texture_mode = LINE_TEXTURE_NONE`)

akien-mga commented on doc/classes/NetworkedMultiplayerPeer.xml:
```diff
@@ -50,10 +50,10 @@
 	</methods>
 	<members>
-		<member name="refuse_new_connections" type="bool" setter="set_refuse_new_connections" getter="is_refusing_new_connections">
+		<member name="refuse_new_connections" type="bool" setter="set_refuse_new_connections" getter="is_refusing_new_connections" default="true">
```
Should be checked, documented default differs from generated one.

bojidar-bg replied:
I think this happens because the default value is different for different subclasses. Here, the one from the GDNative peer was probably taken.

akien-mga replied:
I'm not sure how we should handle such case. If the base implementation has no default value per se (as there is no boolean behind it, and the methods are virtual and implemented), I guess it's a situation where no default value should be retrieved? i.e. we shouldn't check subclasses and just abort if the base class is inconclusive.

CC @Faless

Faless replied:
Well, I assume most `NetworkedMultiplayerPeer` will have the default value of `false` to this (with the exception of GDNative when it has no interface). We can change that value there if we want (have if fail to `false` with no interface).

As Akien said, I'm not sure how useful are defaults for properties based on virtual methods, if you need to have a default there it would make more sense having it in the subclass but that would mean changing the current doc quite a lot.

bojidar-bg replied:
Defaults on virtual classes are useful for stuff like BaseButton. Not sure how to mark properties which depend on virtual methods, as there aren't many such properties...

akien-mga commented on doc/classes/ParticlesMaterial.xml:
```diff
@@ -13,166 +13,166 @@
 			Linear acceleration randomness ratio. Default value: [code]0[/code].
 		</member>
-		<member name="orbit_velocity" type="float" setter="set_param" getter="get_param">
+		<member name="orbit_velocity" type="float" setter="set_param" getter="get_param" default="null">
```
Same issue as for CPUParticles.

akien-mga commented on doc/classes/RigidBody2D.xml:
```diff
@@ -114,64 +114,64 @@
 			The body's total applied torque.
 		</member>
-		<member name="bounce" type="float" setter="set_bounce" getter="get_bounce">
+		<member name="bounce" type="float" setter="set_bounce" getter="get_bounce" default="null">
```
Bug here. (Why does RigidBody2D still have a bounce property though, shouldn't it be in the PhysicsMaterial?).

akien-mga commented on doc/classes/RigidBody2D.xml:
```diff
@@ -114,64 +114,64 @@
 			If [code]true[/code], internal force integration is disabled for this body. Aside from collision response, the body will only move as determined by the [method _integrate_forces] function.
 		</member>
-		<member name="friction" type="float" setter="set_friction" getter="get_friction">
+		<member name="friction" type="float" setter="set_friction" getter="get_friction" default="null">
```
Same here.

akien-mga commented on doc/classes/Spatial.xml:
```diff
@@ -293,29 +293,29 @@
 			The [SpatialGizmo] for this node. Used for example in [EditorSpatialGizmo] as custom visualization and editing handles in Editor.
 		</member>
-		<member name="global_transform" type="Transform" setter="set_global_transform" getter="get_global_transform">
+		<member name="global_transform" type="Transform" setter="set_global_transform" getter="get_global_transform" default="null">
```
This and `rotation` seem bogus.

akien-mga commented on doc/classes/SpatialMaterial.xml:
```diff
@@ -12,255 +12,255 @@
+		<member name="albedo_texture" type="Texture" setter="set_texture" getter="get_texture" default="null">
 		</member>
-		<member name="anisotropy" type="float" setter="set_anisotropy" getter="get_anisotropy">
+		<member name="anisotropy" type="float" setter="set_anisotropy" getter="get_anisotropy" default="null">
```
Several float, bool or int properties are `null` here.

akien-mga committed:
doc: Remove hardcoded default values from descriptions

They are now generated automatically by doctool.

akien-mga committed:
doc: Remove null default values that can't be determined

Applying #30187.

akien-mga committed:
CPUParticles2D: Fix gravity to be the same as in Particles2D and Area2D

This was likely a typo, 10 times the standard gravity is 98.0665
(ca. 98), not 98.8.

akien-mga commented on doc/classes/OS.xml:
```diff
@@ -800,58 +800,58 @@
 	</methods>
 	<members>
-		<member name="clipboard" type="String" setter="set_clipboard" getter="get_clipboard">
+		<member name="clipboard" type="String" setter="set_clipboard" getter="get_clipboard" default="&quot;an editor if it were&quot;">
```
This one gets the clipboard of the system running `doctool` as default value, probably not what we want ;)

akien-mga replied:
Fixed.

akien-mga commented on doc/classes/OS.xml:
```diff
@@ -800,58 +800,58 @@
 			[b]Note:[/b] This property has no effect if [b]Project &gt; Project Settings &gt; Display &gt; Window &gt; Per-pixel transparency &gt; Allowed[/b] setting is disabled.
 		</member>
-		<member name="window_position" type="Vector2" setter="set_window_position" getter="get_window_position">
+		<member name="window_position" type="Vector2" setter="set_window_position" getter="get_window_position" default="Vector2( 0, 436 )">
```
Same here, this was the position of my doctool window and it changes when running doctool again. I assume all other `OS` properties suffer from the same issue, so they're not real default values but those of the editor running doctool.

bojidar-bg replied:
Well, the default value of window position, screen size, etc. is whatever the window manager tell us. Probably they should be masked with ADD_PROPERTY_DEFAULT though.

akien-mga replied:
Done.

akien-mga committed:
Fix uninitialized variables in Line2D, JSONParseResult and TileMap

akien-mga committed:
doc: Mark PhysicsBody bounce/friction as deprecated

akien-mga committed:
doc: Enforce default values for OS properties

Otherwise we end up fetching values from the current OS instance
when running doctool, so they would change based on the system or
even simply due to changes to the system clipboard.
